graph_dfs_debug/graph.py

```diff
@@ -2,39 +2,65 @@
 Simple graph implementation compatible with BokehGraph class.
 """
 class Vertex:
+    # so why is component default value -1?
     def __init__(self, label, component=-1):
         self.label = str(label)
         self.component = component
 
     def __repr__(self):
         return 'Vertex: ' + self.label
 
-    """Trying to make this Graph class work..."""
+# v = Vertex("me")
+# print(v.component)
+# print(v.label)
+# print(v)
+
+    #Trying to make this Graph class work..."""
 class Graph:
     def __init__(self):
+        # so again graph is just a dictionary of sets
         self.vertices = {}
         self.components = 0
 
+    def get_vertex_label(self, vertex):
+        pass
+
     def add_vertex(self, vertex, edges=()):
-        self.vertices[vertex] = set(edges)
+        self.vertices[Vertex(vertex)] = set(edges)
 
     def add_edge(self, start, end, bidirectional=True):
-        self.vertices[start].add(start)
+        # graph is undirected by default
+        start_class = None
+        end_class = None
+        for key in self.vertices.keys():
+            if key.label == start:
+                start_class = key
         if bidirectional:
-            self.vertices[end].add(end)
+            self.add_vertex(end)
+            for key in self.vertices.keys():
+                if key.label == end:
+                    end_class = key
+            self.vertices[start_class].add(end_class)
+            self.vertices[end_class].add(start_class)
+        else:
+            self.vertices[start_class].add(Vertex(end))
 
     def dfs(self, start, target=None):
```

Comment:
We're calling this `dfs` method on line 78 and it's expecting an iterable data structure (set or list) to be returned but your code is returning the last visited node instead.

```
def dfs(self, start, target=None):
    stack = [start]
    visited = set([start])

    while stack:
        current = stack.pop(-1)
        if current == target:
            break
        visited.add(current)
        stack.extend(self.vertices[current] - visited)
        visited.update(self.vertices[current])

    return visited
```